### PR TITLE
obs-studio-plugins.obs-freeze-filter: init at 0.3.3

### PR DESCRIPTION
**Pull request by pschmitt**
## Description of changes

Add package for https://github.com/exeldro/obs-freeze-filter, an OBS Studio plugin to freeze a source.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 251613` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>obs-studio-plugins.obs-freeze-filter</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13873,6 +13873,16 @@
...
+  pschmitt = {
+    email = "philipp@schmitt.co";
+    github = "pschmitt";
+    githubId = 37886;
+    name = "Philipp Schmitt";
+    matrix = "@pschmitt:one.ems.host";
+    keys = [{
+      fingerprint = "9FBF 2ABF FB37 F7F3 F502  44E5 DC43 9C47 EACB 17F9";
+    }];
+  };
```
Comment (matthiasbeyer) [inline maintainers/maintainer-list.nix:13885]: This should be a seperate commit

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  '';
+
+  dontWrapQtApps = true;
+
+  meta = with lib; {
+    description = "Plugin for OBS Studio to freeze a source using a filter ";
+    homepage = "https://github.com/exeldro/obs-freeze-filter";
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+  };
```
Comment (matthiasbeyer) [inline pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix:40]: Would you like to add yourself as maintainer here?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (pschmitt) [reply maintainers/maintainer-list.nix:13885]: Can do. Will do.

Comment (pschmitt) [reply maintainers/maintainer-list.nix:13885]: Done: 13192814b87ccce470a264f6c7a379fedfaf1a7b

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix (hunk 2 above)
Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix]: Yes. Done in 86cb069686fbf851e57570094f0badc91dd785da

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
https://github.com/NixOS/nixpkgs/pull/251613/commits/13192814b87ccce470a264f6c7a379fedfaf1a7b does not  match the [commit message conventions of nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)! 

Please fix that :smile:

**Comment by pschmitt** [comment]
> https://github.com/NixOS/nixpkgs/pull/251613/commits/13192814b87ccce470a264f6c7a379fedfaf1a7b does not  match the [commit message conventions of nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)! 
> 
> Please fix that :smile: 

Done in c8aef93daea3d1f5e7edd2399a80da3b507e87a6

**Comment by pschmitt** [comment]
Rebased again to remove a trailing space in the package description.

**Review by matthiasbeyer — APPROVED** [review]
Diff LGTM

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    mv "$out/data/obs-plugins" "$out/share/obs"
+    rm -rf "$out/obs-plugins" "$out/data"
+  '';
+
+  dontWrapQtApps = true;
+
+  meta = with lib; {
+    description = "Plugin for OBS Studio to freeze a source using a filter";
+    homepage = "https://github.com/exeldro/obs-freeze-filter";
+    license = licenses.gpl2;
```
Comment (emilylange) [inline pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix:38]:
```suggestion
    license = licenses.gpl2Only;
```

`licenses.gpl2` is deprecated.
See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

Deciding between `gpl2Only` and `gpl2Plus` comes down whether upstream has the "or later" section.
E.g. in the header of the files, or explicitly in stated in the readme.

In this case here, it's `gpl2Only` :)

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, obs-studio
+, qtbase
+}:
+
+stdenv.mkDerivation rec {
```
Comment (emilylange) [inline pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix:9]: The `finalAttrs`-pattern is currently preferred over `rec` when using `stdenv.mkDerivation`.
See https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

```patch
diff --git a/pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix b/pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix
index 33e0ee2cbef7..350c9b8ad088 100644
--- a/pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix
+++ b/pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix
@@ -6,13 +6,14 @@
 , qtbase
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "obs-freeze-filter";
   version = "0.3.3";
+
   src = fetchFromGitHub {
     owner = "exeldro";
     repo = "obs-freeze-filter";
-    rev = version;
+    rev = finalAttrs.version;
     sha256 = "sha256-CaHBTfdk8VFjmiclG61elj35glQafgz5B4ENo+7J35o=";
     fetchSubmodules = true;
   };
@@ -39,4 +40,4 @@ stdenv.mkDerivation rec {
     platforms = platforms.linux;
     maintainers = with maintainers; [ pschmitt ];
   };
-}
+})
```

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix (hunk 3 above)
Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix]: Done: 58582672329877bff707a44462d33cbc38285668

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix (hunk 4 above)
Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix]: Done: 1f0e004f2b23f32568370b84046dd99fc1c6c5f4

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  dontWrapQtApps = true;
+
+  meta = with lib; {
+    description = "Plugin for OBS Studio to freeze a source using a filter";
+    homepage = "https://github.com/exeldro/obs-freeze-filter";
+    license = licenses.gpl2Only;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ pschmitt ];
+  };
+}
```
Comment (emilylange) [inline pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix:40]:
```suggestion
})
```

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix (hunk 5 above)
Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix:40]: oh.

Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix]: fc0e6da3781a8eabf960339dffff849d6c3cf508 :facepalm:

Comment (pschmitt) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix]: Sorry for wasting your time for this.

Comment (emilylange) [reply pkgs/applications/video/obs-studio/plugins/obs-freeze-filter.nix]: no worries :)

**Review by emilylange — APPROVED** [review]

**Pull request merged**